src/clojurians_log/views.clj

```diff
@@ -181,7 +181,8 @@
     (list [:div.message
            {:id (cl.tu/format-inst-id inst) :class (when (thread-child? message) "thread-msg")}
            [:a.message_profile-pic {:href (str slack-instance "/team/" slack-id) :style (str "background-image: url(" image-48 ");")}]
-           [:a.message_username {:href (str slack-instance "/team/" slack-id)} name]
+           [:a.message_username {:href (str slack-instance "/team/" slack-id)}
+            (first (filter #(not (clojure.string/blank? %)) [display-name real-name name]))]
```

Comment:
`(some #(when-not (str/blank? %) %) [...])`

Reply:
So I'm guessing `some` will stop as soon as it finds the first match. So it's more efficient?

I did a small test to test this out:
```
Clojure 1.10.1
user=> (some #(do (println %) (when (even? %) %)) [1 2 3 4 5 6])
1
2
2
user=> (first (filter #(do (println %) (not (even? %))) [1 2 3 4 5 6]))
1
2
3
4
5
6
1
```

Reply:
That's interesting, I didn't even think of that.

Clojure does not really have a first class "find" function for getting the first element in a collection for which a predicate is true, the way that for instance Ruby has.

``` ruby
(0..10).find {|x| x.odd?}
```

This is deliberate, there has been discussion about this. The reason is that this always means linearly traversing the collection, which can't be made efficient. The official advice is that in general you should then represent your data in a different way. But of course that's not always possible or desirable. So people end up either doing `(first (filter ...))` or `(some ...)`. I think "some" better communicates the intent, so I consider it more idiomatic, but the downside with `some` is that it returns the result of the function, instead of the element itself, which is why you end up with `#(when ... %)`.

I also wanted to point out to us `str/blank?` instead of `clojure.string` blank. It's more flexible, maybe at some point we want to wrap `clojure.string` in a `lambdaisland.string` namespace, then we can just update the `ns` forms to `[lambdaisland.string :as str]`.

There's a lambda island episode about `some`, it's a surprisingly flexible function, for instance you may see code like this

``` clojure
(if (some #{:foo} coll) ...)
```

or even

```
(if (some (comp #{:foo} val) m) ...)
```

Also a nice combo:

```
(some not-empty coll)
```

Gives you the first thing in `coll` that is not nil and not an empty coll/seq/string.

Reply:
> the downside with some is that it returns the result of the function, instead of the element itself

This was the reason I went with `first filter`, also that it seemed most natural to me: "Take the first element which is not blank"

> I also wanted to point out to use str/blank? 

Yes I completely missed that haha, I'll keep that in mind from now on!

:O `some` does seem to have some interesting use cases, I'll check it in-depth 👍